### PR TITLE
Fixes random id generator for php 7.1+

### PR DESCRIPTION
**Pull request by byrnedo**

### CODE REVIEW

**Comment by byrnedo** [comment]
@repejota Anything you'd like me to change?

File: src/Nats/Connection.php [hunk 1]
```diff
@@ -130,8 +130,12 @@ public function __construct(ConnectionOptions $options = null)
         $this->pubs = 0;
         $this->subscriptions = [];
         $this->options = $options;
-        $randomFactory = new Factory();
-        $this->randomGenerator = $randomFactory->getLowStrengthGenerator();
+        if(version_compare(phpversion(), '7.0', '>')){
+            $this->randomGenerator = new Php71RandomGenerator();
+        } else {
+            $randomFactory = new Factory();
+            $this->randomGenerator = $randomFactory->getLowStrengthGenerator();
```
Comment (dfeyer) [inline src/Nats/Connection.php]: Nitpick, but no needs to create a variable for the factory, this should work

    $this->randomGenerator = (new Factory())->getLowStrengthGenerator();

**Comment by byrnedo** [comment]
If he's going to patch it then great. There wasn't much movement from him at the time I ran into this: https://github.com/ircmaxell/RandomLib/issues/55

**Comment by byrnedo** [comment]
As much as it's nice to hope for an upstream fix for this I would think quite important soon to GTM as 7.1 is getting adopted more and more.

**Comment by repejota** [comment]
🎆

**Pull request merged**